app/reducers/sessions.js

```diff
@@ -0,0 +1,165 @@
+import {
+    NEW_SESSION,
+    SWITCH_SESSION,
+    DELETE_SESSION,
+    UPDATE_CONNECTION,
+    UPDATE_CONFIGURATION,
+    UPDATE_IPC_STATE
+} from '../actions/sessions.js';
+import Immutable from 'immutable';
+import {EMPTY_SESSION} from '../constants/constants';
+
+// see end of file for `list` description
+const INITIAL_STATE = Immutable.Map({
+    list: Immutable.Map({}),
+    sessionSelected: 0,
+    startupIPC: Immutable.Map({})
+});
+
+export default function sessions(state = INITIAL_STATE, action) {
+
+    switch (action.type) {
+
+        /*
+         * TODO: make this action take a new random index
+         * instead of pushing to the list
+         */
+
+        case NEW_SESSION:
+            return state.mergeIn(['list'],
+                {
+                    [action.payload]: {
+                        configuration:
+                            Immutable.Map (EMPTY_SESSION.CONFIGURATION),
+                        connection:
+                            Immutable.Map(EMPTY_SESSION.CONNECTION),
+                        /*
+                         * include startup inter process communcation
+                         * into each session
+                         */
+                        ipc:
+                            state.get('startupIPC')
+                    }
+                }
+            );
+
+        case SWITCH_SESSION:
+            return state.merge({sessionSelected: action.payload});
+
+        case DELETE_SESSION:
+            return state.deleteIn(['list', `${action.payload}`]);
+
+        case UPDATE_CONFIGURATION:
+            return state.mergeIn(
+                ['list', `${state.get('sessionSelected')}`, 'configuration'],
+                action.payload
+            );
+
+        case UPDATE_CONNECTION:
+            return state.mergeIn(
+                ['list', `${state.get('sessionSelected')}`, 'connection'],
+                action.payload
+            );
+
+        case UPDATE_IPC_STATE:
+            if (state.get('list').size > 0) {
+                return state.mergeIn(
+                    ['list', `${state.get('sessionSelected')}`, 'ipc'],
+                    Immutable.fromJS(action.payload)
+                );
+            } else {
+                return state.mergeIn(
+                    ['startupIPC'],
+                    Immutable.fromJS(action.payload)
+                );
+            }
```

Comment:
Instead of having `startupIPC`, can we just initialize an entry in the `INITIAL_STATE`?

Something like

```
const INITIAL_STATE = Immutable.Map({
     list: Immutable.Map({['0']: {}}),
     sessionSelected: '0'
 });
```